r2-shared/Publication/Locator.ts

```diff
@@ -0,0 +1,34 @@
+interface LocatorText {
+  after?: string;
+  before?: string;
+  highlight?: string;
+}
+
+interface StartEndInfo {
+  cssSelector: string;
+  textNodeIndex: number;
+  charOffset?: number;
+}
+
+interface DomRangeInfo {
+  start: StartEndInfo;
+  end?: StartEndInfo;
+}
+
+interface Locations {
+  fragments?: Array<string>;
+  progression?: number;
+  position?: number;
+  totalProgression?: number;
+  cssSelector?: string;
+  partialCfi?: string;
+  domRange?: DomRangeInfo;
+}
+
+export interface Locator {
+  href: string;
+  type: string;
+  title?: string;
+  locations?: Locations;
+  text?: LocatorText;
```

Comment:
To simplify, on mobile these properties are not nullable. This way we don't have to guard for null.